build_projects/update-dependencies/UpdateFilesTargets.cs

```diff
@@ -87,7 +87,24 @@ public static BuildTargetResult ReplaceProjectJson(BuildTargetContext c)
         {
             List<DependencyInfo> dependencyInfos = c.GetDependencyInfos();
 
-            IEnumerable<string> projectJsonFiles = Directory.GetFiles(Dirs.RepoRoot, "project.json", SearchOption.AllDirectories);
+            string[] sourceDirsToUpdate =
+            {
+                "build_projects",
+                "pkg",
+                // Don't upgrade RTM versions in InternalAbstractions and DependencyModel.
+                // See https://github.com/dotnet/core-setup/issues/275
+                //"src",
+                "test",
+                "tools"
+            };
+
+            IEnumerable<string> projectJsonFiles = sourceDirsToUpdate
+                .SelectMany(name =>
+                    Directory.GetFiles(
+                        Path.Combine(Dirs.RepoRoot, name),
+                        "project.json",
+                        SearchOption.AllDirectories))
+                .ToArray();
```

Comment:
Pointing out new logic I added to avoid upgrading `src` project.jsons. I tested it out, and it produces the exact upgrades that are in this PR.